lms/djangoapps/course_home_api/outline/views.py

```diff
@@ -423,33 +400,18 @@ def save_course_goal(request):  # pylint: disable=missing-function-docstring
     if not course_id:
         raise ParseError("'course_id' is required.")
 
-    if COURSE_GOALS_NUMBER_OF_DAYS_GOALS.is_enabled():
-        # If body doesn't contain the required goals fields, return 400 to client.
-        if subscribed_to_reminders is None:
-            raise ParseError("'subscribed_to_reminders' is required.")
-
-        try:
-            add_course_goal(request.user, course_id, subscribed_to_reminders, days_per_week)
-            return Response({
-                'header': _('Your course goal has been successfully set.'),
-                'message': _('Course goal updated successfully.'),
-            })
-        except Exception:
-            raise UnableToSaveCourseGoal  # pylint: disable=raise-missing-from
-
-    else:
-        # If body doesn't contain 'goal', return 400 to client.
-        if not goal_key:
-            raise ParseError("'goal_key' is required.")
-
-        try:
-            add_course_goal_deprecated(request.user, course_id, goal_key)
-            return Response({
-                'header': _('Your course goal has been successfully set.'),
-                'message': _('Course goal updated successfully.'),
-            })
-        except Exception:
-            raise UnableToSaveCourseGoal  # pylint: disable=raise-missing-from
+    # If body doesn't contain the required goals fields, return 400 to client.
+    if subscribed_to_reminders is None:
+        raise ParseError("'subscribed_to_reminders' is required.")
+
+    try:
+        add_course_goal(request.user, course_id, subscribed_to_reminders, days_per_week)
+        return Response({
+            'header': _('Your course goal has been successfully set.'),
+            'message': _('Course goal updated successfully.'),
+        })
+    except Exception:
+        raise UnableToSaveCourseGoal  # pylint: disable=raise-missing-from
```

Comment:
You can fix this pylint disable by doing:
```
except Exception as exc:
    raise UnableToSaveCourseGoal from exc
```

Reply:
That is lovely. Thanks